File: Tests/Tests/LoginPageTests.cs
using Atata;
using NUnit.Framework;
using Tests.Pages;

namespace Tests.Tests
{
    [TestFixture]
    public class LoginPageTests : BaseTest
    {
        [Test]
        public void RegisterAndLoginTest() =>
            Go.To<LoginPage>()
                .OpenRegistration()
                .Register("Login", "test-test@test.com", "PaSsWoRd")
                .Login("login", "PaSsWoRd")
                .PageTitle.Should.Equal("Deposit calculator");

        [TestCase("test", "newyork2")]
        [TestCase("text", "newyork1")]
        [TestCase("test ", "newyork1")]
        [TestCase("test", "newyork1 ")]
        [TestCase("", "")]
        [TestCase("test", "")]
        [TestCase("", "newyork1")]
        public void LoginWithInvalidCredentialsTest(string login, string password) =>
            CreateDefaultUser()
                .Login(login, password)
                .Error
                .Content.ExpectTo.Equal("Incorrect user name or password!");
    }
}

File: Tests/tests/LoginPageTests.py
import unittest

from Constants import Constants
from pages.LoginPage import LoginPage
from tests.BaseTest import BaseTestCase


class LoginPageTests(BaseTestCase):
    def test_empty_login_and_password(self): self.negative("", "")
    def test_empty_login(self):              self.negative("", Constants.Password)
    def test_empty_password(self):           self.negative(Constants.Login, "")
    def test_valid_only_login(self):         self.negative(Constants.Login, Constants.Password + "!")
    def test_valid_only_password(self):      self.negative(Constants.Login + "!", Constants.Password)
    def test_valid_login_and_password(self): self.positive(Constants.Login, Constants.Password)

    def negative(self, login, password):
        login_page = LoginPage(self.driver)
        error = login_page.login(login, password)

        self.assertEqual(error, "Incorrect user name or password!")

    def positive(self, login, password):
        login_page = LoginPage(self.driver)
        login_page.login(login, password)

        self.assertEqual(self.driver.title, "Deposit calculator")


if __name__ == '__main__':
    unittest.main()
